### ISSUE TITLE
gRPC version upgrade

### ISSUE THREAD

**Issue by jabubake** [GoogleCloudPlatform/google-cloud-java#1776]
Looks like certain version of Maven (2.2.1, 3.2.2 and 3.2.3) run into issues with com.google.code.gson:gson:2.3: https://github.com/google/gson/issues/588
The issue is resolved in gson : 2.3.1.
The older version(2.3) is linked in google-cloud-core from grpc 1.0.3. 
 grpc 1.1+ support gson:2.7. Is there a reason not to be on atleast 1.1.x of grpc ?

 [INFO] |  |  +- io.grpc:grpc-protobuf:jar:1.0.3:compile
[INFO] |  |  |  +- (com.google.guava:guava:jar:19.0:compile - omitted for duplicate)
[INFO] |  |  |  +- com.google.protobuf:protobuf-java-util:jar:3.0.2:compile
[INFO] |  |  |  |  +- (com.google.protobuf:protobuf-java:jar:3.0.2:compile - omitted for conflict with 3.0.0)
[INFO] |  |  |  |  +- (com.google.guava:guava:jar:18.0:compile - omitted for conflict with 20.0)
[INFO] |  |  |  |  \- com.google.code.gson:gson:jar:2.3:compile

**Comment by buchgr** [comment]
**tl;dr;** 
Yes to upgrading! :-)

There is another problem due to no fault of Google Cloud Java. gRPC 1.1.x and 1.0x aren't binary compatible (on a Java Byte Code level).

This commit [1] changed the return type of `NettyChannelBuilder.build()` from `ManagedChannelImpl` (1.0.x) to `ManagedChannel` (>= 1.1.0). A Java method signature does not include the return type, however, a Java Byte Code method signature does include the return type. So one can have methods of the same name, but with a different return type. And that's the problem.

For example, this code [2] calls `NettyChannelBuilder.build()` with return type `ManagedChannelImpl` when compiled against gRPC 1.0.3. However, gRPC >= 1.1.0 only has `NettyChannelBuilder.build()` with return type `ManagedChannel` and thus running this code fails with a `NoSuchMethodException`. 

Here is the byte code version of `SpannerOptions$NettyRpcChannelFactory#newChannel(String, int)` with the class file taken from maven central version 0.10.0-beta. Notice, how it calls a method with a `ManagedChannelImpl` return type.

![grpc breakage](https://cloud.githubusercontent.com/assets/1388179/24142337/8e72e3bc-0e26-11e7-864a-24e9e7c3a4b4.png)


[1] https://github.com/grpc/grpc-java/commit/d17a7b5bd47ff7eb0f47f64ff82d6cb2457a662a#diff-34c34e0485cd6a3ebabb1f46de341a1aL269
[2] https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java#L278

**Comment by garrettjonesgoogle** [comment]
We already have this issue for upgrading grpc: https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1691 ; work hasn't started yet.

We have also already pulled in a change in gax-java (https://github.com/googleapis/gax-java/issues/208) to switch from `NettyChannelBuilder` to `ManagedChannelBuilder`. This is in master but hasn't been released yet.

**Comment by garrettjonesgoogle** [comment]
Resolving this issue in favor of #1691 .
